mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -248,22 +249,23 @@ impl Deserializable for BlockHeader {
         let solution = if version == 4 { Some(reader.read_list()?) } else { None };
 
         // https://en.bitcoin.it/wiki/Merged_mining_specification#Merged_mining_coinbase
-        let aux_pow = if version == AUX_POW_VERSION_DOGE || version == AUX_POW_VERSION_SYS {
-            let coinbase_tx = deserialize_tx(reader, TxType::StandardWithWitness)?;
-            let parent_block_hash = reader.read()?;
-            let coinbase_branch = reader.read()?;
-            let blockchain_branch = reader.read()?;
-            let parent_block_header = Box::new(reader.read()?);
-            Some(AuxPow {
-                coinbase_tx,
-                parent_block_hash,
-                coinbase_branch,
-                blockchain_branch,
-                parent_block_header,
-            })
-        } else {
-            None
-        };
+        let aux_pow =
+            if version == AUX_POW_VERSION_DOGE || version == AUX_POW_VERSION_SYS || version == AUX_POW_VERSION_NMC {
```

Comment:
I usually use `if` like you, but I just had an idea to use `matches` macro here.
```rust
let aux_pow = if matches!(version, AUX_POW_VERSION_DOGE | AUX_POW_VERSION_SYS | AUX_POW_VERSION_NMC) {
  ...
}
```

Reply:
Nice, and I think this is less verbose. 👍

Reply:
do you think it's required to do this for other cases like this in `block_header`?

Reply:
I think it would be great!

Reply:
done